models/repo_generate.go

```diff
@@ -39,6 +40,34 @@ func (gro GenerateRepoOptions) IsValid() bool {
 	return gro.GitContent || gro.Topics || gro.GitHooks || gro.Webhooks || gro.Avatar || gro.IssueLabels // or other items as they are added
 }
 
+func parseGiteaTemplate(tmpDir string) ([]glob.Glob, error) {
+	globs := make([]glob.Glob, 0)
+	gtPath := filepath.Join(tmpDir, ".giteatemplate")
+	if _, err := os.Stat(gtPath); err != nil {
+		return globs, err
+	}
+	content, err := ioutil.ReadFile(gtPath)
+	if err != nil {
+		return globs, err
+	}
+
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		line = strings.TrimSpace(line)
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		g, err := glob.Compile(line, '.', '/')
```

Comment:
I'm not so sure about using `'.'` as token separator. I believe it's there for internet addresses like `www.gitea.com`, not for file paths. In files I think it will equate `/` to `.` (granted those are edge cases):

* `**Makefile` will match `subdir/Makefile` but also `subdir/Docs.Makefile`
* `css**` will match `css/a`, `css/b` but also `css.x/a` and `css.zop/xxx`, and _not match_ `cssx/a` or `csszop/xxx`.

Whereas using only `'/'` as separator:

* `**Makefile` will match `subdir/Makefile` and not `subdir/Docs.Makefile`
* `css**` will match `css/a` and `css/b`, and won't match any of `css.x/a`, `css.zop/xxx`, `cssx/a` or `csszop/xxx`

But I might have that wrong. Jeez, I get dizzie with this glob syntax. I'd take RegExp anytime. 😵

Reply:
This now uses only '/' separator.